nixos/modules/config/users-groups.nix

```diff
@@ -635,15 +638,13 @@ in {
           content = "${base64}${sep}${base64}";
           mcf = "^${sep}${scheme}${sep}${content}$";
         in
-        if (user.hashedPassword != null   # login disabled
+        if (allowsLogin user.hashedPassword
             && user.hashedPassword != ""  # login without password
             && builtins.match mcf user.hashedPassword == null)
```

Comment:
I think you already do these two checks with `allowsLogin`.

Reply:
If you mean the empty string, it's different: it's a special value but it does allow login, so it's not in `allowsLogin`.

Reply:
So It should be in the `allowsLogin` check, right?